### PR TITLE
D4CHAT-44 feat: Dockerfile and docker-compose for the app

### PR DESCRIPTION
**Pull request by GrigoriiBerezin**
Error with filled pool of connections for now
17:14:11.591 [HikariPool-1 housekeeper] trace_id= user_id= DEBUG com.zaxxer.hikari.pool.HikariPool HikariPool-1 - Fill pool skipped, pool is at sufficient level.
17:14:41.592 [HikariPool-1 housekeeper] trace_id= user_id= DEBUG com.zaxxer.hikari.pool.HikariPool HikariPool-1 - Pool stats (total=10, active=0, idle=10, waiting=0)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM sbtscala/scala-sbt:eclipse-temurin-jammy-19.0.1_10_1.8.2_3.2.2
+
+WORKDIR /d4-chat
+ADD . /d4-chat
+
+EXPOSE ${APP_PORT}
+
+CMD ["sbt", "run"]
```
Comment (AlekseiLitkovetc) [inline Dockerfile:8]: It's not a good idea to run the app via sbt on a server due to the additional overhead and resource usage

Comment (AlekseiLitkovetc) [inline Dockerfile:8]: The final image weight will be huge - around 1GB.
We can use docker stages to reduce the final size in times - the final weight will be around 285Mb.

The Dockerfile might looks like this:
```
FROM sbtscala/scala-sbt:openjdk-17.0.2_1.8.0_3.2.1 as build
RUN mkdir -p /root/build/project
COPY ./ /root/build
WORKDIR /root/build
RUN sbt d4-chat/stage

FROM openjdk:11-jre-slim-bullseye
WORKDIR .
COPY --from=build /root/build/target/universal/stage/ /app/
CMD app/bin/main
EXPOSE ${APP_PORT}
```

**Review by AlekseiLitkovetc — CHANGES_REQUESTED** [review]

File: build.sbt [hunk 2]
```diff
@@ -49,6 +49,9 @@ lazy val root = project
     scalafmtLogOnEachError := true,
     scalafixOnCompile      := true
   )
+  .settings(
+    Compile / run / mainClass := Some("ru.d4team.chat.Main")
```
[the remark this answers is not in the document]
Comment (GrigoriiBerezin) [reply build.sbt:53]: We have 2 objects that extend `ZioAppDefault` and `sbt` doesn't know which one to start

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+EXPOSE ${APP_PORT}
+
+ENV APP_HOST=${APP_HOST} \
+    APP_PORT=${APP_PORT} \
+    DB_HOST=${DB_HOST} \
+    DB_PORT=${DB_PORT} \
+    DB_NAME=${DB_NAME} \
+    DB_USER=${DB_USER} \
+    DB_SCHEMA=${DB_SCHEMA} \
+    DB_PASSWORD=${DB_PASSWORD}
```
Comment (AlekseiLitkovetc) [inline Dockerfile:20]: Does it set env variables for a final image?

If yes, can we override them when we want to run a compouse file on different environment?

Comment (GrigoriiBerezin) [reply Dockerfile:20]: Yes, it will be overwrited

**Review by AlekseiLitkovetc — APPROVED** [review]

**Review by PatokovN — APPROVED** [review]

**Review by OlegKoshelev — APPROVED** [review]

**Pull request merged**